externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Conversion/TcpToTosa/TcpToTosa.h

```diff
@@ -0,0 +1,20 @@
+#ifndef TORCH_MLIR_DIALECTS_CONVERSION_TCPTOTOSA_TCPTOTOSA_H_
+#define TORCH_MLIR_DIALECTS_CONVERSION_TCPTOTOSA_TCPTOTOSA_H_
+
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+
+#define GEN_PASS_DECL_CONVERTTCPTOTOSA
```

Comment:
Why do we need this GEN_PASS_DECL_CONVERTTCPTOTOSA? I don't think we do this in the main Torch-MLIR codebase. I think we just have a single PassDetail.h which only need to be included by the .cpp files.

Reply:
Aah okay. Thanks for pointing that out.

Reply:
@silvasean `GEN_PASS_CLASSES` [has been recently deprecated](https://discourse.llvm.org/t/psa-deprecation-warning-per-pass-autogenerated-macros-and-pass-options/64998), and will be removed in the near future. `GEN_PASS_DECL_PASSNAME` is the recommended way to approach this moving forward. E.g. see [this MLIR-HLO commit](https://github.com/tensorflow/mlir-hlo/commit/7596fd322d2408bf88d4c0e2748d95a3f760aefe) for an example.

By the way, functions like `createConvertTcpToTosaPass` can be autogenerated, but only if you don't specify `let constructor` in Passes.td, e.g. see [this StableHLO PR](https://github.com/openxla/stablehlo/pull/176) for an example.

Reply:
Oh neat. I wasn't aware of that.

Reply:
> By the way, functions like createConvertTcpToTosaPass can be autogenerated, but only if you don't specify let constructor in Passes.td, e.g. see https://github.com/openxla/stablehlo/pull/176 for an example.

Good to know that. Thanks.